src/Main.elm

```diff
@@ -1,47 +1,129 @@
 module Main exposing (..)
 
 import Browser
-import Html exposing (Html, text, div, h1, img)
-import Html.Attributes exposing (src)
+import Html exposing (..)
+import Html.Attributes exposing ( href, class )
+import Html.Events exposing (..)
+import Http
+import RemoteData exposing (RemoteData, WebData)
+import Json.Decode as Decode exposing (Decoder, int, list, string)
+import Json.Decode.Pipeline exposing (required)
 
 
 ---- MODEL ----
 
+type alias News =
+    { id: Int }
+
 
 type alias Model =
-    {}
+    { hackerNews : WebData (List News) }
 
+emptyModel : Model
+emptyModel =
+    { hackerNews = RemoteData.Loading }
 
 init : ( Model, Cmd Msg )
 init =
-    ( {}, Cmd.none )
+    ( emptyModel, Cmd.none )
 
 
 
 ---- UPDATE ----
 
 
 type Msg
-    = NoOp
+    = DataReceived (WebData (List News))
+
+dataDecoder: Decoder News
+dataDecoder =
```

Comment:
@Diama1 

This decoder says I expect to get an object with structure [id:123], but what you get is 123.
Consider decoding the int, and then, making it `News`.
[RemoteData.andThen](https://package.elm-lang.org/packages/krisajenkins/remotedata/latest/RemoteData#andThen) can help you here.